zerver/tests/test_outgoing_webhook_system.py

```diff
@@ -210,6 +211,43 @@ def deactivate_bot_first(*args: Any) -> None:
 
         self.assertEqual(bot_owner_notification.recipient_id, bot_user.bot_owner.recipient_id)
 
+    def test_notify_bot_owner_with_deactivated_owner(self) -> None:
+        bot_user = self.example_user("outgoing_webhook_bot")
+        mock_event = self.mock_event(bot_user)
+        service_handler = GenericOutgoingWebhookService("token", bot_user, "service")
+
+        def deactivate_bot_owner_first(*args: Any) -> None:
+            bot_user = get_user_profile_by_id(args[0]["user_profile_id"])
+            assert bot_user.bot_owner is not None
+            do_deactivate_user(bot_user.bot_owner, acting_user=None)
+            notify_bot_owner(*args)
+
+        expected_fail = mock.patch(
+            "zerver.lib.outgoing_webhook.notify_bot_owner", side_effect=deactivate_bot_owner_first
+        )
+
+        with mock.patch.object(
+            service_handler, "session"
+        ) as session, expected_fail as mock_fail, self.assertLogs(level="WARNING") as m:
+            session.post.return_value = ResponseMock(408)
+            final_response = do_rest_call("", mock_event, service_handler)
+            assert final_response is not None
+
+            self.assertEqual(
+                m.output,
+                [
+                    f'WARNING:root:Message http://zulip.testserver/#narrow/stream/999-Verona/topic/Foo/near/ triggered an outgoing webhook, returning status code 408.\n Content of response (in quotes): "{final_response.text}"'
```

Comment:
Something is wrong with this error message: We shouldn't have a `near:` link without a message ID. Do you understand the bug?

Reply:
Hmm, I am not sure why us it so. This test is adapted from a similar test just above.

Reply:
We have `id` as empty string in the message content for `mock_event`.